ts/src/base/Exchange.ts

```diff
@@ -1359,6 +1429,52 @@ export default class Exchange {
     // ------------------------------------------------------------------------
     // METHODS BELOW THIS LINE ARE TRANSPILED FROM JAVASCRIPT TO PYTHON AND PHP
 
+    setExchangePropAllCase (key: string, value: any = undefined): any {
+        this[key] = value;
+        const unCamelCasedKey = this.unCamelCase (key);
+        this[unCamelCasedKey] = value;
+    }
+
+    checkProxySettings () {
+        const proxyUrl = (this.proxyUrl !== undefined) ? this.proxyUrl : this.proxy_url;
+        const proxyUrlCallback = (this.proxyUrlCallback !== undefined) ? this.proxyUrlCallback : this.proxy_url_callback;
+        // for backwards compatibility,added old keys too
+        let proxyHttp = (this.proxyHttp !== undefined) ? this.proxyHttp : this.proxy_http;
+        // support for backward compatibility (note, atm safeStringN does not work in python for class https://app.travis-ci.com/github/ccxt/ccxt/builds/263490790#L4765 , so we have to use separate safeString)
+        if (proxyHttp === undefined) {
+            const httpProxy = this.getProperty (this, 'httpProxy');
```

Comment:
Same thing here, let's replace this line with an explicit check
```
const httpProxy = this.httpProxy;
```

Reply:
that was done